datanode/main.go

```
@@ -0,0 +1,92 @@
+// datanode/main.go
+package main
+
+import (
+	"log/slog"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"strings"
+
+	"github.com/ZhenbangYou/TinyDFS/tiny-dfs/common"
+)
+
+// Command Line Args:
+// Args[1]: Namenode endpoint (IP:port)
+// Args[2]: Datanode endpoint (IP:port)
+// Args[3]: Optional log file path
+func main() {
+	// Check command line arguments
+	if len(os.Args) < 3 || len(os.Args) > 4 {
+		slog.Error("expect 3 or 4 command line arguments", "actual argument count", len(os.Args))
+		os.Exit(1)
```

panicking instead of logging+exit